src/mlpack/methods/ann/brnn.hpp

```diff
@@ -76,6 +77,35 @@ class BRNN
        MergeOutputType mergeOutput = MergeOutputType(),
        InitializationRuleType initializeRule = InitializationRuleType());
 
+/**
+   * Check if the optimizer has MaxIterations() parameter, if it does
+   * then check if it's value is less than value of number of
+   * datapoints in the dataset.
+   *
+   * @tparam OptimizerType Type of optimizer to use to train the model.
+   * @param optimizer optimizer used in the training process.
+   * @param no_of_datapoints Number of datapoints in the dataset.
+   */
+  template<typename OptimizerType>
+  typename std::enable_if<
+      HasMaxIterations<OptimizerType, size_t&(OptimizerType::*)()>
+      ::value, void>::type
+  WarnMessage(OptimizerType& optimizer, size_t no_of_datapoints) const;
```

Comment:
Please use camel casing for all names, also what about using `samples` instead of `no_of_datapoints` here?

Reply:
Thanks for pointing this out @zoq I have incorporated this.